packages/parser/src/language/sankey/sankey.langium

```diff
@@ -0,0 +1,42 @@
+grammar Sankey
+
+// common stuff in ../common/common.langium
+// can't import directly because of SANKEY_LINK_SOURCE parse rule
+fragment TitleAndAccessibilities:
+  ((accDescr=ACC_DESCR | accTitle=ACC_TITLE | title=TITLE) NEWLINE+)+
+;
+
+terminal NEWLINE: /\r?\n/;
+terminal ACC_DESCR: /[\t ]*(?:accDescr(?:[\t ]*:[\t ]*[^\n\r]*?(?=%%)|\s*{[^}]*})|accDescr(?:[\t ]*:[\t ]*[^\n\r]*|\s*{[^}]*}))/;
+terminal ACC_TITLE: /[\t ]*(?:accTitle[\t ]*:[\t ]*[^\n\r]*?(?=%%)|accTitle[\t ]*:[\t ]*[^\n\r]*)/;
+terminal TITLE: /[\t ]*(?:title(?:[\t ]+[^\n\r]*?|)(?=%%)|title(?:[\t ]+[^\n\r]*|))/;
+hidden terminal WHITESPACE: /[\t ]+/;
+hidden terminal YAML: /---[\t ]*\r?\n[\s\S]*?---[\t ]*(?!.)/;
+hidden terminal DIRECTIVE: /[\t ]*%%{[\s\S]*?}%%\s*/;
+hidden terminal SINGLE_LINE_COMMENT: /[\t ]*%%[^\n\r]*(\r?\n)*/;
+
+interface Sankey {
+  accDescr?: string;
+  accTitle?: string;
+  title?: string;
+  links: SankeyLink[];
+  nodes: string[];
+}
+
+// actual grammar
+entry Sankey returns Sankey:
+  NEWLINE*
+  "sankey-beta"
+  (
+    NEWLINE* TitleAndAccessibilities links+=SankeyLink+
+    | NEWLINE+ links+=SankeyLink+
+  )
+;
+
+SankeyLink:
+  source=SANKEY_LINK_SOURCE target=SANKEY_LINK_TARGET value=SANKEY_LINK_VALUE NEWLINE+
+;
+
+terminal SANKEY_LINK_VALUE returns number: /,[\t ]*(0|[1-9][0-9]*)(\.[0-9]+)?/;
+terminal SANKEY_LINK_TARGET: /,[\t ]*("[^\n\r"]+"|[^\n\r,]+)/;
+terminal SANKEY_LINK_SOURCE: /sankey-link-source/;
```

Comment:
Is this a temporary plug?

Reply:
Short answer: no.

Long answer: If we set the actual rule in grammar it won't allow us to generate AST because it interferes with keyword rule. The same issue occurs with grammars that matches everything, e.g., `timeline`:

> ...and add rules manually in the correct order, then reintegrate the custom rule back again, which worked fine for all rules, but the keyword won't match now:
> 
> ```
> Error: Errors detected in definition of Lexer:
> Token: ->timeline<- can never be matched.
> Because it appears AFTER the Token Type ->TIMELINE_PERIOD_TITLE<-in the lexer's definition.
> See https://chevrotain.io/docs/guide/resolving_lexer_errors.html#UNREACHABLE
> ```
> https://github.com/mermaid-js/mermaid/pull/4450#issuecomment-1659314287

Reply:
It is very strange workaround for the problem. From what I've read in the comment it must be resolved by reordering patterns.

Reply:
Reordering can be done if the AST has been generated, but that error was thrown when trying to generate the AST using `langium-cli`.

> You'll probably need to set the pattern explicitly in the token builder.
> https://github.com/mermaid-js/mermaid/pull/4450#issuecomment-1660067236